bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/provider/ShellyChannelDefinitions.java

```diff
@@ -126,13 +126,14 @@ public ShellyChannelDefinitions(@Reference ShellyTranslationProvider translation
                 // Device
                 .add(new ShellyChannel(m, CHGR_DEVST, CHANNEL_DEVST_NAME, "deviceName", ITEMT_STRING))
                 .add(new ShellyChannel(m, CHGR_DEVST, CHANNEL_DEVST_GATEWAY, "gatewayDevice", ITEMT_STRING))
-                .add(new ShellyChannel(m, CHGR_DEVST, CHANNEL_DEVST_ITEMP, "deviceTemp", ITEMT_TEMP))
+                .add(new ShellyChannel(m, CHGR_DEVST, CHANNEL_DEVST_ITEMP, "system:indoor-temperature", ITEMT_TEMP))
```

Comment:
All these changes you do to use system channel types concerns only channels created dynamically ?
No need for upgrade instructions ?

What about the old channel types ? If they are no more used, you should remove them.

Reply:
There are only a very few channels, which are created statically on the device level (see device.xml), all others are created dynamically.

I tried to remove the old channel definitions, but that leads to the situation that the framework blocks starting the binding for 60sec trying to find the definitions somewhere. So, it's the hen and the egg. Old definitions are still required when starting the new binding version doing the channel upgrades and then they are no longer required on next startup. Again, including upgrade information for all channels for all groups and devices would create a mess with a ton of duplicate overhead.

Reply:
@lolodomo could we mark this one as resolved?

Reply:
@J-N-K could you advice how such a Thing upgrade case should be handled?

Reply:
@J-N-K Any idea how to best deal with Thing upgrades in the case of dynamically created channels?

Reply:
Except for this and the other upgrade comment htis PR is ready to merge. Woudl be nice if someone with more experience could comment if this is ok or not, hint: @J-N-K :-)